### PR TITLE
Improved node value formatter; added NodeDefs functions

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: src/category/categoryItems.ts [hunk 1]
```diff
@@ -4,10 +4,22 @@ import { CategoryItem } from './item'
 
 const getCode = (item: CategoryItem): string => item.props.code ?? ''
 
-const getLabelOrCode = (item: CategoryItem, lang: LanguageCode) =>
-  Strings.defaultIfEmpty(getCode(item))(item.props.labels?.[lang])
+const getLabel = (item: CategoryItem, lang: LanguageCode): string => item.props.labels?.[lang] ?? ''
```
Comment (persas) [inline src/category/categoryItems.ts:7]: These three functions seem similar, maybe is worth combining them and getting the output based on a parameter

File: src/node/nodeValueFormatter.ts [hunk 2]
```diff
@@ -1,63 +1,88 @@
...
 import { Taxa } from '../taxonomy'
-import { DateFormats, Dates, Numbers, Objects } from '../utils'
+import { DateFormats, Dates, Numbers, Objects, Strings } from '../utils'
+import { Node } from './node'
 import { NodeValues } from './nodeValues'
 
-const formatters: { [key in NodeDefType]?: any } = {
-  [NodeDefType.code]: (params: { survey: Survey; value: any; showLabel: boolean; lang: LanguageCode }) => {
-    const { survey, value, showLabel, lang } = params
+const extractCategoryItem = (params: { survey: Survey; node?: Node; value: any }) => {
```
Comment (persas) [inline src/node/nodeValueFormatter.ts:10]: Maybe we can also type the output: 

```typescript
const extractCategoryItem = (params: { survey: Survey; node?: Node; value: any }): CategoryItem | null => {
```

Comment (SteRiccio) [reply src/node/nodeValueFormatter.ts:10]: right, good catch

File: src/category/categoryItems.ts (hunk 1 above)
Comment (SteRiccio) [reply src/category/categoryItems.ts:7]: how can the generic function be named?

**Review by persas — APPROVED** [review]

**Pull request merged**